plugin/pkg/utils/miscellaneous.go

```diff
@@ -102,3 +103,129 @@ func ValidateInternalDomainChangeForSeed(oldSeedSpec, newSeedSpec *core.SeedSpec
 
 	return nil
 }
+
+// ValidateDefaultDomainsChangeForSeed returns an error when default domains that are being used by a shoot scheduled on the seed are removed,
+// or when default domains are being added but do not cover all the globally available default domains used by shoots on the seed.
+func ValidateDefaultDomainsChangeForSeed(oldSeedSpec, newSeedSpec *core.SeedSpec, seedName string, shootLister gardencorev1beta1listers.ShootLister, secretLister kubecorev1listers.SecretLister, kind string) error {
+	oldDomains := sets.New[string]()
+	for _, defaultDomain := range oldSeedSpec.DNS.Defaults {
+		oldDomains.Insert(defaultDomain.Domain)
+	}
+
+	newDomains := sets.New[string]()
+	for _, defaultDomain := range newSeedSpec.DNS.Defaults {
+		newDomains.Insert(defaultDomain.Domain)
+	}
+
+	shoots, err := shootLister.List(labels.Everything())
```

Comment:
You can make use of label selector `name.seed.gardener.cloud/<seed-name>=true`
```suggestion
	shoots, err := shootLister.List(labels.SelectorFromSet(labels.Set{v1beta1constants.LabelPrefixSeedName + seedName: "true"}))
```

Reply:
Other functions in the package use `labels.Everything()`. I do not feel comfortable making this change with this PR. If done it should be introduced for all similar functions 🤔